crypto/rand/rand_lib.c

```diff
@@ -178,6 +178,8 @@ size_t rand_drbg_get_entropy(RAND_DRBG *drbg,
                                    prediction_resistance,
                                    NULL, 0) != 0)
                 bytes = bytes_needed;
+            drbg->reseed_next_counter
+                = tsan_load(&drbg->parent->reseed_prop_counter);
```

Comment:
Something isn't making sense. You're just going to overwrite this with drbg->reseed_prop_counter below.

Reply:
Below is in the else block where the case parent==NULL is handled.
That is where the counter gets incremented, here we just follow the parent.
It is important too use the parent->reseed_prop_counter before
calling rand_drbg_unlock(drbg->parent).